FF152 Relnote: with  { type: "text" } supported (#44257)
mdn/content: files/en-us/mozilla/firefox/experimental_features/index.md

## JavaScript

### Text module import

The `with` clause [`{ type: "text" }`](/en-US/docs/Web/JavaScript/Reference/Statements/import/with#text_modules_type_text) allows importing a module's source as a string value.

> The import ignores the media type of the response: the content is parsed as text even if the source contains scripts or other executable code.

Author: ```suggestion
The media type of the response is ignored, and the content is parsed as text even if the source contains scripts or other executable code.
```

Reviewer: Nice!

## JavaScript

### Text module import

Support for the `with` clause [`{ type: "text" }`](/en-US/docs/Web/JavaScript/Reference/Statements/import/with#text_modules_type_text).

> This ensures that a module served with a media type of `text/plain` is imported as text into a string value, even if it contains JavaScript or some other format.

Reviewer: - The first sentence is incomplete. But both these can be combined:
```suggestion
The `with` clause [`{ type: "text" }`](/en-US/docs/Web/JavaScript/Reference/Statements/import/with#text_modules_type_text) allows importing a module's source as a string value, even if the source contains JavaScript code.
```

- "a module served with a media type of `text/plain`" - as per the docs, the request is sent with `Accept: text/plain`, but the response Content-Type is not checked

Author: Slight modification

```suggestion
The `with` clause [`{ type: "text" }`](/en-US/docs/Web/JavaScript/Reference/Statements/import/with#text_modules_type_text) allows importing a module's source as a string value.
```

> If served with a different media type the import fails.

Reviewer: This doesn't seem accurate as per docs at [import/with#text_modules_type_text](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import/with#text_modules_type_text) that say that "the value of the response's [Content-Type](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Content-Type) header is ignored, and all files are parsed as UTF-8"....so the import doesn't fail if served with a different media type

Author: Thanks. YOu are correct. Took me ages to find confirmation, but its in https://github.com/whatwg/html/pull/11933

> The type: 'text' import attribute is introduced here without any MIME type requirement, and without any validity or well-formedness checks on the returned string value....

```suggestion
The import ignores the media type of the response: the content is parsed as text even if the source contains scripts or other executable code.
```
